core/src/main/java/org/mule/runtime/core/internal/processor/strategy/TransactionAwareStreamEmitterProcessingStrategyDecorator.java

```diff
@@ -104,14 +106,17 @@ public ReactiveProcessor onPipeline(ReactiveProcessor pipeline) {
     return pub -> subscriberContext()
         .flatMapMany(ctx -> {
           if (isTxActive(ctx)) {
-            // The profiling events related to the processing strategy scheduling are triggered independently of this being
+            // The profiling events related to the processing strategy scheduling are triggered no matter this being
             // a blocking processing strategy that does not involve a thread switch.
-            return buildFlux(pub)
+            ReactorPublisherBuilder<Flux<CoreEvent>> reactorPublisherBuilder = buildFlux(pub)
                 .profileProcessingStrategyEvent(location, getDataProducer(PS_SCHEDULING_FLOW_EXECUTION), artifactId, artifactType)
                 .profileProcessingStrategyEvent(location, getDataProducer(STARTING_FLOW_EXECUTION), artifactId, artifactType)
                 .transform(BLOCKING_PROCESSING_STRATEGY_INSTANCE.onPipeline(pipeline))
-                .profileProcessingStrategyEvent(location, getDataProducer(FLOW_EXECUTED), artifactId, artifactType)
-                .build();
+                .profileProcessingStrategyEvent(location, getDataProducer(FLOW_EXECUTED), artifactId, artifactType);
+            if (featureFlags.isEnabled(ENABLE_PROFILING_SERVICE)) {
```

Comment:
are we 100% sure we want to use the feature flag mechanism for this? the FF were originally design as a way to maintain backward compatibility when fixing things.

Reply:
We currently working in the implementation of a better mechanism for this. The idea is to refactor the places where we ask for this feature flag in order to start using such mechanism as soon as it is ready.